passes/pass4.nim

```diff
@@ -213,11 +218,22 @@ proc colorGraph(gr: var Graph) =
             doAssert not e.noCopy, "cannot satisfy constraints"
           else:
             gr.nodes[e.dst].color = color
+        elif e.noCopy and gr.nodes[e.dst].color != color:
+          # problem: the source and destination should have the same color,
+          # they but don't. Register an ordered conflict pair
+          var
+            a = color
+            b = gr.nodes[e.dst].color
+          if b < a:
+            swap(a, b)
```

Comment:
```suggestion
          let
            a = min(color, gr.nodes[e.dst].color)
            b = max(color, gr.nodes[e.dst].color)
```

Soft suggestion: it drops the `var` and I think is roughly as readable; also it's consistent with the `min` that comes later

Reply:
Good call on using a more functional style. Instead of using `min`/`max`, I've added a dedicated procedure for constructing the sorted pair, which I think conveys the intent a bit better.